### PR TITLE
Adding PPOM middleware

### PR DESCRIPTION
**Pull request by jpuri**
Adding PPOMMiddleware

### CODE REVIEW

File: src/ppom-middleware.test.ts [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+      storageBackend: storageBackendReturningData,
+      provider: { sendAsync: Promise.resolve() },
+      chainId: '0x1',
+      onNetworkChange: (_callback) => undefined,
+    });
+    const middlewareFunction = createPPOMMiddleware(ppomController);
+    expect(middlewareFunction).toBeDefined();
+  });
+
+  describe('PPOMMiddleware', () => {
```
Comment (matthewwalsh0) [inline src/ppom-middleware.test.ts:36]: I'm not sure I follow what this grouping is, if all these tests are verifying the logic of `createPPOMMiddleware`.

File: src/ppom-middleware.test.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    it('should call ppomController.use when invoked', async () => {
+      const useMock = jest.fn();
+      const controller = {
+        use: useMock,
+      };
+      const middlewareFunction = createPPOMMiddleware(controller as any);
+      await middlewareFunction({}, undefined, () => undefined);
+      expect(useMock).toHaveBeenCalledTimes(1);
```
Comment (matthewwalsh0) [inline src/ppom-middleware.test.ts:44]: Do we need a test to verify the `ppomResponse` property is set on the original request?

File: src/ppom-middleware.test.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+          callback(ppom);
+        },
+      };
+      const middlewareFunction = createPPOMMiddleware(controller as any);
+      const nextMock = jest.fn();
+      await middlewareFunction({}, undefined, nextMock);
+      expect(nextMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call next method when ppomController.use throws error', async () => {
```
[the remark this answers is not in the document]
Comment (jpuri) [reply src/ppom-middleware.test.ts:62]: Yes @matthewwalsh0 : UI will show error if validation by PPOM fails.

File: src/ppom-middleware.test.ts (hunk 2 above)
Comment (jpuri) [reply src/ppom-middleware.test.ts:44]: Yep that is good idea, I will add the test.

File: src/ppom-middleware.test.ts (hunk 1 above)
Comment (jpuri) [reply src/ppom-middleware.test.ts:36]: good point, let me re-group

**Review by matthewwalsh0 — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Pull request merged**